corehq/apps/app_manager/models.py

```diff
@@ -4468,10 +4468,32 @@ def make_build(self, comment=None, user_id=None):
         assert copy._id
         prune_auto_generated_builds.delay(self.domain, self._id)
 
-        self.check_build_dependencies(new_build=copy)
+        self.analyse_app_build(new_build=copy)
 
         return copy
 
+    def analyse_app_build(self, new_build):
+        self.check_for_custom_callouts(new_build)
+        self.check_build_dependencies(new_build)
+
+    def check_for_custom_callouts(self, new_build):
+        from corehq.apps.app_manager.util import app_callout_templates
+
+        templates = next(app_callout_templates)
+        template_ids = set([t['id'] for t in templates])
+
+        def app_has_custom_intents():
+            return any(
+                any(set(form.wrapped_xform().odk_intents) - template_ids)
+                for form in new_build.get_forms()
+            )
+
+        if app_has_custom_intents():
+            metrics_counter(
+                'commcare.app_build.custom_app_callout',
+                tags={'domain': new_build.domain, 'app_id': new_build.copy_of},
```

Comment:
Just checking on this, same as the other PR, does adding tags effect cost on datadog and how? I assume a new custom metric is a new cost of its own.

Reply:
I see there is context about this in the description.

So, we are saying since making new version happens infrequently and that we are reporting only when its applicable this should be okay, seems fair.

I see Jenny already noticed this PR, but Just tagging @gherceg as well for visibility that a new custom tag is being added.
(let me know if its not necessary for us to notify you or someone everytime a new tag is added)

Reply:
I still want to check out how often a new build is created so we can at least have a sense of frequency. Will update when I know.